sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -57,8 +59,21 @@ case class ScalaUDF(
       children: Seq[Expression],
       inputTypes: Seq[DataType],
       udfName: Option[String]) = {
-    this(
-      function, dataType, children, inputTypes, udfName, nullable = true, udfDeterministic = true)
+    this(function, dataType, children, inputTypes, udfName,
+      nullable = true, udfDeterministic = true, nullableTypes = Nil)
+  }
+
+  // Constructor from Spark 2.3
```

Comment:
I'm not sure we want to keep doing this. cc @gatorsmile

Reply:
By convention, everything under catalyst package is private, so compatibility is not a concern here.